### PR TITLE
feat(ui): new user experience

### PR DESCRIPTION
**Pull request by maryhipp**
## Summary

Is a user has no images, we assume they are a new user and show them details on what to do

Screenshot 2024-09-19 at 4 27 48 PM

## Checklist

- [ ] _The PR has a short but descriptive title, suitable for a changelog_
- [ ] _Tests added / updated (if applicable)_
- [ ] _Documentation added / updated (if applicable)_

### CODE REVIEW

File: invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+        if (hasBoards) {
+            if (boardList.filter(board => board.image_count > 0).length > 0) {
+                return true
+            }
+        }
+        return uncategorizedImages ? uncategorizedImages.total > 0 : true
+
+
+    }, [boardList, uncategorizedImages, loadingBoards, loadingImages])
```
[the remark this answers is not in the document]
Comment (maryhipp) [reply invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts:30]: @psychedelicious ☝️

[the remark this answers is not in the document]
Comment (psychedelicious) [reply invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts:30]: Ya I think this is as good as it gets

File: invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        offset: 0,
+        limit: 0,
+        is_intermediate: false,
+    })
+
+    const hasImages = useMemo(() => {
+        // default to true
+        if (loadingBoards || loadingImages) {
+            return true
+        }
```
Comment (psychedelicious) [inline invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts:18]: One improvement might be to return a sentinel value while loading and use that to display a loading indicator:

```ts
export const LOADING_SYMBOL = Symbol("LOADING")

export const useHasImages = () => {
    const { data: boardList, isLoading: loadingBoards } = useListAllBoardsQuery({ include_archived: true })
    const { data: uncategorizedImages, isLoading: loadingImages } = useListImagesQuery({
        board_id: "none",
        offset: 0,
        limit: 0,
        is_intermediate: false,
    })

    const hasImages = useMemo(() => {
        // default to true
        if (loadingBoards || loadingImages) {
            return LOADING_SYMBOL
        }

...
```

Then in `<NoContentForViewer />`:

```tsx
if (hasImages === LOADING_SYMBOL) {
   return <Loading />
}
```

Comment (maryhipp) [reply invokeai/frontend/web/src/features/gallery/hooks/useHasImages.ts]: Nice, added

**Review by psychedelicious — APPROVED** [review]

**Pull request merged**